### ISSUE TITLE
lint: port checks from go-doccheck

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#470]
We can port some doc comment checks from https://github.com/Quasilyte/go-doccheck.

I thought to do it as a separate checker, but now that we also check for comments in some checkers, it could be a good idea to merge it here as well.

Checkers:
1. `boolFuncComment`: bool functions comment style ("reports whether ...").
2. `commentDoubleSpace`: no double space after a period. (unimplemented in go-doccheck.)
3. `commentPunct`: comment ends with punctuation. But should not trigger for inline code snippets.

There could be more.
It's also probably a good idea to create separate issue for every checker proposed above.

**Comment by cristaloleg** [comment]
2. commentDoubleSpace - isn't this handled by gofmt?
3. commentPunct - golint?

**Comment by Quasilyte** [comment]
Nope.

**Comment by Quasilyte** [comment]
I have some distaste in comments like this:
```
// For each input parameter to a function, the escapeReturnEncoding describes
// how the parameter may leak to the function's outputs. This is currently the
// "level" of the leak where level is 0 or larger (negative level means stored into
// something whose address is returned -- but that implies stored into the heap,
// hence EscHeap, which means that the details are not currently relevant. )
```
Note the last sentence:
1. The period is inside the parenthesis. Should be outside. Or the sentence it clarifies itself should end with a period, then the whole clarification can go into parenthesis with period **inside** parenthesis.
2. There is a space between punctuation and `)`. I think that does not comply to typography standards either.

This may be going too deep with this. We're not going to implement proper natural language processing and neither will use existing engines (fat dependency, but it could be optional).

**Comment by cristaloleg** [comment]
> This may be going too deep with this.
it's a quite simple example that can be found by a small regexp, so why not? 😉

**Comment by Quasilyte** [comment]
Even more pedantic:
```
const (
	bitsPerOutputInTag = 3                                 // For each output, the number of bits for a tag
	bitsMaskForTag     = uint16(1<<bitsPerOutputInTag) - 1 // The bit mask to extract a single tag.
	maxEncodedLevel    = int(bitsMaskForTag - 1)           // The largest level that can be stored in a tag.
)
```
2 last single-line comments end with a period while the first one does not.
On a global scale, it's hard to enforce either of these, but for gendecls (like const specs) it's possible.

Example below should either have no periods or have them in all 3 places.
